worksheet_D/oxo.py

```diff
@@ -1,88 +1,176 @@
-            continue
-
-        return x, y
-
-
-# The main game. You should not need to edit this.
-if __name__ == '__main__':
-    board = OxoBoard()
-    current_player = 1
-    while True:
-        board.show()
-        print "Choose a square, Player", current_player
-        x, y = input_square()
-
-        if board.set_square(x, y, current_player):
-            # Move was played successfully, so check for a winner
-            winner = board.get_winner()
-            if winner != 0:
-                print "Player", winner, "wins!"
-                break   # End the game
-            elif board.is_board_full():
-                print "It's a draw!"
-                break   # End the game
-            else:
-                # Switch players
-                if current_player == 1:
-                    current_player = 2
-                else:
-                    current_player = 1
-        else:
-            # Move was not played successfully
-            print "That square is already filled!"
+class OxoBoard:
+    def __init__(self, x, y):
+
+        # create a 2D array of a 3 X 3 grid
+        self.board = [[0, 0, 0],[0, 0, 0],[0, 0, 0]]
+
+    def get_square(self, x, y):
+        #this function returns the value of a square in the array
+        return self.board[x][y]
+
+
+    def set_square(self, x, y, mark):
+        #This function sets a square in the array
+        if self.get_square(x,y) == 0:
+            self.board[x][y] = mark
+            return True
+        else:
+            return False
+
+
+    def is_board_full(self):
+
+        fullTiles = 0
+
+        for i in range(len(self.board)):
+            for j in range(len(self.board[i])):
+                squareContent = self.get_square(i, j)
+                if squareContent != 0:
+                    fullTiles += 1
+                    if fullTiles == 9:
+                        return True
+                else:
+                    return False
+
+
+
+    def get_winner(self, latestX, latestY, currentPlayer):
+
+        total = latestX + latestY
+        checkDiagnols = 0
+        rowCount = 0
+        coloumnCount = 0
+
+        # Check if we need to check diagnols by checking if the sum of the x and y tiles is even
+
+        if total % 2 == 0:
+            checkDiagnols == 1
+
+        ### CHECK HORIZONTAL AND VERTICAL 3 IN A ROW ###
+
+        for i in range(3):
+
+
+            Square = self.get_square(i, latestY)
+            if Square == currentPlayer:
+                rowCount += 1
+                if rowCount == 3:
+                    return currentPlayer
+
+
+            Square = self.get_square(latestX, i)
+            if Square == currentPlayer:
+                coloumnCount += 1
+                if coloumnCount == 3:
+                    return currentPlayer
+
+        ###### CHECK DIAGNOLS ######
+
+        # here i try to speed up the process by eliminating useless checks
+
+        if checkDiagnols == 1:
+            if latestX and latestY == 1:

```

Comment:
I don't think this does what you want it to -- it evaluates to
```
if (latestX) and (latestY == 1)
```
which in Python is the same as
```
if (latestX != 0) and (latestY == 1)
```